studio/components/interfaces/Billing/InvoiceStatusBadge.tsx

```diff
@@ -0,0 +1,89 @@
+import { FC } from 'react'
+import * as Tooltip from '@radix-ui/react-tooltip'
+import { Badge } from 'ui'
+
+interface Props {
+  status: string
+}
+
+export const invoiceStatusMapping: Record<string, { label: string; badgeColor: string }> = {
+  draft: {
+    label: 'Upcoming',
+    badgeColor: 'yellow',
+  },
+  paid: {
+    label: 'Paid',
+    badgeColor: 'green',
+  },
+  void: {
+    label: 'Forgiven',
+    badgeColor: 'green',
+  },
+
+  // We do not want to overcomplicate it for the user, so we'll treat uncollectible/open the same from a user perspective
+  // it's an outstanding invoice
+  uncollectible: {
+    label: 'Outstanding',
+    badgeColor: 'red',
+  },
+  open: {
+    label: 'Outstanding',
+    badgeColor: 'red',
+  },
+}
+
+const InvoiceStatusBadge: FC<Props> = ({ status }) => {
+  const statusMapping = invoiceStatusMapping[status]
+
+  return (
+    <Tooltip.Root delayDuration={0}>
+      <Tooltip.Trigger>
+        <Badge
+          size="small"
+          className="capitalize"
+          // @ts-ignore
+          color={statusMapping?.badgeColor || 'gray'}
+        >
+          {statusMapping?.label || status}
+        </Badge>
+      </Tooltip.Trigger>
+      <Tooltip.Content side="bottom">
+        <Tooltip.Arrow className="radix-tooltip-arrow" />
+        <div
+          className={[
+            'rounded bg-scale-100 py-1 px-2 leading-none shadow',
+            'w-[300px] space-y-2 border border-scale-200',
+          ].join(' ')}
+        >
+          {statusMapping?.label === 'Outstanding' && (
```

Comment:
just a nit really - could just use status === 'xxx' as the check here

and probably move the different statuses into an enum (but we can refactor this next time too)

```
export enum InvoiceStatus = {
  DRAFT: 'draft',
  PAID: 'paid,
 ...etc
}
```

Reply:
Also typed the invoices in the state, so they are no longer `any`